src/app/components/generic-event-page/generic-event-page.component.html

```diff
@@ -26,7 +26,9 @@ <h1>{{ title }}</h1>
                 </ng-template>
             </div>
 
-            <app-generic-filter class="generic-filter" [showHiddenSection]="showFilterCheckboxes"></app-generic-filter>
+            <app-generic-filter (sortData)="sortEvents($event)" (filtersData)="filterEvents($event)" [showHiddenSection]="showHiddenSection()">
```

Comment:
`showHiddenSection()`
put the result of the function in a variable and bind that one in the template.